# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /Users/domi/NetBeansProjects/m2release/plugin
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: src/main/java/org/jvnet/hudson/plugins/m2release/M2ReleaseAction.java
--- src/main/java/org/jvnet/hudson/plugins/m2release/M2ReleaseAction.java Base (BASE)
+++ src/main/java/org/jvnet/hudson/plugins/m2release/M2ReleaseAction.java Locally Modified (Based On LOCAL)
@@ -26,6 +26,7 @@
 import hudson.maven.MavenModule;
 import hudson.maven.MavenModuleSet;
 import hudson.model.Action;
+import hudson.model.BuildBadgeAction;
 import hudson.model.Hudson;
 
 import java.io.IOException;
@@ -203,7 +204,7 @@
 		
 		// schedule release build
 		synchronized (project) {
-			if (project.scheduleBuild(0, new ReleaseCause())) {
+			if (project.scheduleBuild(0, new ReleaseCause(), new M2ReleaseAction.ReleaseIconAction())) {
 				m2Wrapper.enableRelease();
 				m2Wrapper.setVersions(versions);
 				m2Wrapper.setAppendHudsonBuildNumber(appendHudsonBuildNumber);
@@ -230,4 +231,29 @@
 	private String getString(String key, Map<?,?> httpParams) {
 		return (String)(((Object[])httpParams.get(key))[0]);
 	}
+
+
+    /**
+     * Enables displaying of the release icon next to the build.
+     * TODO: At some time this could also contain the actual version build.
+     */
+    public static class ReleaseIconAction implements BuildBadgeAction {
+
+
+        public ReleaseIconAction() {
 }
+
+        public String getIconFileName() {
+            return null;
+        }
+
+        public String getDisplayName() {
+            return null;
+        }
+
+        public String getUrlName() {
+            return null;
+        }
+
+    }
+}
Index: src/main/resources/org/jvnet/hudson/plugins/m2release/M2ReleaseAction/ReleaseIconAction/badge.jelly
--- src/main/resources/org/jvnet/hudson/plugins/m2release/M2ReleaseAction/ReleaseIconAction/badge.jelly Locally New
+++ src/main/resources/org/jvnet/hudson/plugins/m2release/M2ReleaseAction/ReleaseIconAction/badge.jelly Locally New
@@ -0,0 +1,5 @@
+<j:jelly xmlns:j="jelly:core">
+        <img width="16" height="16"
+             title="This is a Relsease Build"
+             src="${imagesURL}/16x16/package.gif"/>
+</j:jelly>  
\ No newline at end of file
